<launch>

 <!-- joy node -->
  <node respawn="false" pkg="joy" type="joy_node" name="joy" >
    <!-- <param name="dev" type="string" value="/dev/input/js0" /> -->
    <!-- <param name="deadzone" value="0.12" /> -->
    <param name="autorepeat_rate" value="10.0" />
  </node>

 <!-- Axes -->

  <node pkg="vrep_vsv_driver" type="teleop_base.py" name="teleop_base" output="screen">
      <param name="axis_linear" value="1" type="int"/>
      <param name="axis_angular" value="0" type="int"/>
      <param name="scale_linear" value="1.0" type="double"/>
      <param name="scale_angular" value="0.7" type="double"/>
      <remap from="/teleop_base/joy" to="/joy"/>
      <remap from="/teleop_base/twistCommand" to="/vsv_driver/twistCommand"/>
  </node>

  <node pkg="vrep_vsv_driver" type="vsv_arm_ik.py" name="arm_ik" output="screen">
      <param name="vsv_name" value="VSV"/>
      <param name="max_velocity" value="0.5"/>
      <remap from="/arm_ik/joint_command" to="/VSV/aggregated/command"/>
      <remap from="/arm_ik/joint_state" to="/VSV/aggregated/state"/>
  </node>

  <node pkg="vrep_vsv_driver" type="teleop_geom.py" name="teleop" output="screen">
      <param name="vsv_name" value="VSV"/>
      <param name="axis_arm_x" value="3"/>
      <param name="axis_arm_y" value="6"/>
      <param name="axis_arm_z" value="4"/>
      <param name="arm_velocity" value="0.25"/>
      <param name="home_button" value="1"/>
      <param name="ready_button" value="0"/>
      <param name="move_button" value="2"/>
      <remap from="/teleop/joy" to="/joy"/>
      <remap from="/teleop/position_command" to="/arm_ik/position"/>
      <remap from="/teleop/twist_command" to="/arm_ik/twist"/>
      <remap from="/teleop/tool_command" to="/arm_ik/tool_orientation"/>
  </node>

  <node pkg="vrep_vsv_driver" type="vsv_driver.py" name="vsv_driver" output="screen">
      <param name="vsv_name" value="VSV"/>
      <param name="min_radius" value="5.0"/>
  </node>

  <node pkg="vrep_vsv_driver" type="vsv_arm.py" name="vsv_arm" output="screen">
      <param name="vsv_name" value="VSV"/>
  </node>

</launch>
